serve/entrypoint.py

```diff
@@ -13,21 +13,120 @@
+        cosmos_container = database.get_container_client(
+            config["monitoring_store"]["container_name"]
+        )
+    
+    # If running locally - pull model + datasets from local AML
+    # If running in hosting - pull from shared registry
+    if is_local():
+        ml_client_registry = MLClient(credential=credential,
+                        resource_group_name=config["aml"]["resource_group_name"],
+                        subscription_id=config["subscription_id"],
+                        workspace_name=config["aml"]["workspace_name"])
+    else:
+        ml_client_registry = MLClient(credential=credential,
+                        registry_name=config["registry"]["name"],
+                        registry_location=config["registry"]["location"])
+    
+    # loop the models in config and download them from the registry
+    if "models" in config:
+        for model in config["models"]:
+            logging.info(f'Downloading model {model["name"]}, version {model["version"]}')
+
+            # download the model, clearing out the old dir first
+            download_path = f'{assets_dir}/models/{model["name"]}/{model["version"]}'
+            ensure_path(download_path)
+
+            ml_client_registry.models.download(
+                name=model["name"], 
+                version=model["version"], 
+                download_path=download_path
+            )
+            
+            loaded_models.append(mlflow.pyfunc.load_model(f'{download_path}/{model["name"]}/mlflow-model'))
+
+    # loop the datasets in config and download them from the registry
+    if "datasets" in config:
+        for dataset in config["datasets"]:
+            logging.info(f'Downloading dataset {dataset["name"]}, version {dataset["version"]}')
+
+            # download the model, clearing out the old dir first
+            download_path = f'{assets_dir}/datasets/{dataset["name"]}/{dataset["version"]}'
+            ensure_path(download_path)
 
-def run(model_inputs: dict = None):
+            ds = ml_client_registry.data.get(
+                name=dataset["name"], 
+                version=dataset["version"]
+            )
+
+            data = ml_client_registry.data.get(name=dataset["name"], version=dataset["version"])
+            artifact_utils.download_artifact_from_aml_uri(
+                uri = data.path, 
+                destination = download_path,
+                datastore_operation=ml_client_registry.datastores)
+
+
+def run(config, user_inputs: dict = None):
     # TODO: Add code here that calls your model
     #       model_inputs is a dictionary containing any inputs that were passed to the model endpoint
     #       This function should return a dictionary containing the model results
     
-    logging.info("Model run started")
-    model_results = {"result": "Hello World!"}
-    logging.info("Model run completed")
+    # get feature data
+    query = """SELECT top 10 [csn]
```

Comment:
Can this just reference the same dummy sql script held at dataset_creation/query.sql

Reply:
i've added another sample inference sql script, and load that in instead of the inline SQL. I think the query for inferencing would be different to the one for training wouldn't it?